mmaction/datasets/ava_dataset.py

```diff
@@ -0,0 +1,241 @@
+    ``img_key`` (in format of ``{video_id},{timestamp}``). Example of a pickle
+    file:
+
+    .. code-block:: JSON
+
+        {
+            ...
+            '0f39OWEqJ24,0902':
+                array([[0.011   , 0.157   , 0.655   , 0.983   , 0.998163]]),
+            '0f39OWEqJ24,0912':
+                array([[0.054   , 0.088   , 0.91    , 0.998   , 0.068273],
+                       [0.016   , 0.161   , 0.519   , 0.974   , 0.984025],
+                       [0.493   , 0.283   , 0.981   , 0.984   , 0.983621]]),
+            ...
+        }
+
+    Args:
+        ann_file (str): Path to the annotation file like
+            ``ava_{train, val}_{v2.1, v2.2}.csv``.
+        exclude_file (str): Path to the excluded timestamp file like
+            ``ava_{train, val}_excluded_timestamps_{v2.1, v2.2}.csv``.
+        pipeline (list[dict | callable]): A sequence of data transforms.
+        label_file (str): Path to the label file like
+            ``ava_action_list_{v2.1, v2.2}.pbtxt`` or
+            ``ava_action_list_{v2.1, v2.2}_for_activitynet_2019.pbtxt``.
+            Default: None.
+        filename_tmpl (str): Template for each filename.
+            Default: 'img_{:05}.jpg'.
+        proposal_file (str): Path to the proposal file like
+            ``ava_dense_proposals_{train, val}.FAIR.recall_93.9.pkl``.
+            Default: None.
+        data_prefix (str): Path to a directory where videos are held.
+            Default: None.
+        test_mode (bool): Store True when building test or validation dataset.
+            Default: False.
+        modality (str): Modality of data. Support 'RGB', 'Flow'.
+                        Default: 'RGB'.
+        num_max_proposals (int): Max proposals number to store. Default: 1000.
+        timestamp_start (int): The start point of included timestamps. The
+            default value is referred from the official website. Default: 902.
+        timestamp_end (int): The end point of included timestamps. The
+            default value is referred from the official website. Default: 1798.
+    """
+
+    _FPS = 30
+    _NUM_CLASSES = 81
+
+    def __init__(self,
+                 ann_file,
+                 exclude_file,
+                 pipeline,
+                 label_file=None,
+                 filename_tmpl='img_{:05}.jpg',
+                 proposal_file=None,
+                 data_prefix=None,
+                 test_mode=False,
+                 modality='RGB',
+                 num_max_proposals=1000,
+                 timestamp_start=902,
+                 timestamp_end=1798):
+        self.exclude_file = exclude_file
+        self.label_file = label_file
+        self.proposal_file = proposal_file
+        self.filename_tmpl = filename_tmpl
+        self.num_max_proposals = num_max_proposals
+        self.timestamp_start = timestamp_start
+        self.timestamp_end = timestamp_end
+        self.logger = get_root_logger()
+        super().__init__(
```

Comment:
why put super.init in the middle?

Reply:
we should first assign some inner attributes for `super().__init`

Reply:
why not put them in the argument list?

Reply:
They are commonly not used for `BaseDataset`

Reply:
if they are not used, then why they are required to be assigned beforehand?

Reply:
https://github.com/open-mmlab/mmaction2/blob/master/mmaction/datasets/base.py#L61-L62, and we rewrite the `self.load_annotations()`.

Reply:
ok. add a line of comment explaining this